### PR TITLE
adds basic server and service setup

### PR DESCRIPTION
**Pull request by xh3b4sd**
Towards https://github.com/giantswarm/giantswarm/issues/3160.

### CODE REVIEW

File: flag/flag.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+import (
+	"github.com/giantswarm/microkit/flag"
+
+	"github.com/giantswarm/node-health/flag/service"
+)
+
+type Flag struct {
+	Service service.Service
+}
```
Comment (xh3b4sd) [inline flag/flag.go]: Adding flags. Note, these are not intitialized nor used yet.

File: server/endpoint/endpoint.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	// Dependencies.
+	Logger  micrologger.Logger
+	Service *service.Service
+}
+
+// Endpoint is the endpoint collection.
+type Endpoint struct {
+	Healthz *healthz.Endpoint
+	Version *version.Endpoint
+}
```
Comment (xh3b4sd) [inline server/endpoint/endpoint.go]: Adding endpoints.

File: server/server.go [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+// Config represents the configuration used to create a new server object.
+type Config struct {
+	Logger  micrologger.Logger
+	Service *service.Service
+	Viper   *viper.Viper
+
+	ProjectName string
+}
+
+type Server struct {
```
Comment (xh3b4sd) [inline server/server.go]: Adding the server.

File: service/healthz/health.go [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+	return h, nil
+}
+
+func (h *Healthz) GetHealthz(ctx context.Context) (healthz.Response, error) {
+	r := healthz.Response{
+		Description: Description,
+		Failed:      false,
+		Message:     SuccessMessage,
+		Name:        Name,
+	}
```
Comment (xh3b4sd) [inline service/healthz/health.go]: Adding the health endpoint. It is not doing more for now. Will be extended later on.

File: service/service.go [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+	Description string
+	Flag        *flag.Flag
+	GitCommit   string
+	Name        string
+	Source      string
+	Viper       *viper.Viper
+}
+
+type Service struct {
```
Comment (xh3b4sd) [inline service/service.go]: Wire the services.

File: Gopkg.toml [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+# [prune]
+#   non-go = false
+#   go-tests = true
+#   unused-packages = true
+
+
+
+[[constraint]]
+  branch = "master"
+  name = "github.com/giantswarm/microendpoint"
```
Comment (xh3b4sd) [inline Gopkg.toml]: Setting up some necessary vendor deps.

**Review by kopiczko — APPROVED** [review]

**Pull request merged**